x-pack/plugins/apm/public/components/alerting/ui_components/chart_preview/index.tsx

```diff
@@ -64,36 +67,89 @@ export function ChartPreview({
     opacity: thresholdOpacity,
   };
 
+  const NUM_SERIES = 5;
+  const DEFAULT_DATE_FORMAT = 'Y-MM-DD HH:mm:ss';
+
+  const tooltipProps: TooltipProps = {
+    headerFormatter: ({ value }) => {
+      const dateFormat =
+        (uiSettings && uiSettings.get(UI_SETTINGS.DATE_FORMAT)) ||
+        DEFAULT_DATE_FORMAT;
+      return moment(value).format(dateFormat);
+    },
+  };
+
+  const filteredSeries = useMemo(() => {
+    const sortedSeries = series.sort((a, b) => {
+      const aMax = Math.max(...a.data.map((point) => point.y as number));
+      const bMax = Math.max(...b.data.map((point) => point.y as number));
+      return bMax - aMax;
+    });
+    return sortedSeries.slice(0, NUM_SERIES);
+  }, [series]);
+
+  const barSeries = useMemo(() => {
+    return filteredSeries.reduce<
+      Array<{ x: number; y: Maybe<number>; groupBy: string | undefined }>
+    >((acc, serie) => {
+      const barPoints = serie.data.reduce<
+        Array<{ x: number; y: Maybe<number>; groupBy: string | undefined }>
+      >((pointAcc, point) => {
+        return [...pointAcc, { ...point, groupBy: serie.name }];
+      }, []);
+      return [...acc, ...barPoints];
+    }, []);
+  }, [filteredSeries]);
```

Comment:
Shouldn't this happen on the server? It looks like potentially 1000 groups are sent to the client, while only 3 are rendered.

Having this on the server will also make it possible to be covered by the API tests.

Reply:
Moved this code to server.